pisek/jobs/parts/solution.py

```diff
@@ -229,31 +256,99 @@ def _jobs_points(self) -> list[float]:
             )
         )
 
-    def __str__(self) -> str:
-        s = "("
-        previous = list(
-            map(
-                lambda x: x.verdict if x else None,
-                self._job_results(self.previous_jobs),
+    def status(
```

Comment:
Woudn't it be more readable if this were multiple methods split by verbosity? All the different variants share is code to wrap the predecessor summary in parentheses.
This would also avoid the awkwardness needed when calling this method to determine whether the resulting text should be displayed inline or on multiple lines.